impl/src/main/java/org/ehcache/internal/events/CacheEventNotificationListenerServiceProviderImpl.java

```diff
@@ -44,9 +51,11 @@ public void stop() {
 
   public <K, V> CacheEventNotificationService<K, V> createCacheEventNotificationService(Store<K, V> store) {
     ThreadPoolsService threadPoolsService = serviceProvider.findService(ThreadPoolsService.class);
+    TimeSourceConfiguration timeSourceConfig = findSingletonAmongst(TimeSourceConfiguration.class, config);
```

Comment:
This is never going to work. The config you get will never be a `TimeSourceConfiguration` and so this always returns null.
I believe you found a shortcut in the service system where we just use a configuration for the time source and not a service. I do not see any way of making this work without fixing that issue.
I agree that the `TimeSourceService` will be very shallow and similar to the `TimeSourceConfiguration` but we have a system in place to access services not service configs.
One good way of testing what I am saying is an integration test that tries to configure a different time source. You will see it never makes it here.